### PR TITLE
Restore USB circular writes and fix images not getting deleted in order.

### PR DESCRIPTION
**Pull request by mazar-farran**
Ticket: https://linear.app/hivemapper/issue/IMCO-166/investigate-hdc-recording-issue

Check for gnss date and use it as an offset if it's available.

Related to: https://github.com/Hivemapper/odc-api/pull/221

### CODE REVIEW

**Review by punov — APPROVED** [review]

File: dashcam/package/camera-node/camera-node.mk
Comment (punov) [reply dashcam/package/camera-node/camera-node.mk:7]: @mazar-farran The only callout is that I think we need to include the progress of that branch to the build:
https://github.com/Hivemapper/hdc_firmware/pull/257
This is the actual latest 4.8.10 release, and I can merge it to main, if you don't mind

Comment (mazar-farran) [reply dashcam/package/camera-node/camera-node.mk:7]: Sure sounds good, I'll rebase on top of that once it's merged to main

**Comment by mazar-farran** [comment]
Hi @punov 

This was not going to work prior to the latest commit.  Only the folder names were fixed, not the image names.  Also, the folder was not updating immediately because we were waiting 5 minutes after a date change to update the folder name.

Also, I noticed there's no circular write logic, unless I'm missing something.  Is this right? In that case how could the user have lost images in the middle of the trip?

**Comment by mazar-farran** [comment]
Tested outside and saw dates being written correctly to the correct folder. Tested usb removal inside and it worked. Tested write limit = 20 and removals seem to work.

**Comment by mazar-farran** [comment]
Do not merge yet, this does not handle full USBs correctly. Will update tomorrow and rerelease as 4.8.18

**Comment by mazar-farran** [comment]
Updated to correctly check for usb space remaining.  Upticked version to 4.8.18

**Review by punov — DISMISSED** [review]

**Review by punov — APPROVED** [review]

**Pull request merged**
